src/jit/compiler.cpp

```diff
@@ -540,7 +540,7 @@ var_types    Compiler::argOrReturnTypeForStruct(unsigned size, CORINFO_CLASS_HAN
         break;
 
     default:
-#if FEATURE_MULTIREG_RET
+#if FEATURE_MULTIREG_RET && !defined(_TARGET_X86_)
```

Comment:
I am current working on implementing the FEATURE_MULTIREG_RET for ARM64

When FEATURE_MULTIREG_RET is defined for a target the value of MAX_RET_MULTIREG_BYTES should also be updated.  This value is just the maximum size that we would ever see  for any struct that is returned in multiple registers, so for x86 that would probably be 8 (or maybe 0).  Setting this to 8 doesn't mean that all 8 byte structs or other 8 byte types have to be returned in registers.  It would just mean that there is some 8 byte type or struct that is returned in registers.  I believe that TYP_LONG on x86 is return in two registers, but 8 byte structs are not returned in two registers.

As the code already does the proper thing here, simply adding a comment about x86 behavior is sufficient:

// For _TARGET_X86_ we never pass or return a TYP_STRUCT type using multiple registers.